### DISCUSSION TITLE
Filter data provider with empty string

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#9367] [Help]
Hello!

I am having difficulty with what should be a simple operation. Let me explain in a few words:
I have a data provider that loads a SQL table, on this table I would like to filter the elements that do not have a first name for example (empty string).

Filter with : First name is empty 
=> Does not return any results because the filter uses the null value

Filter with : First name is equal to value ____ (empty field)
=> Returns all results (while the query does 'equal name null')?

It should be easy to make a filter like this, right? Filter on an empty string?

Thank you!

ps: I can of course write an SQL query but it adds a lot of steps for something basic

**Comment by andz-bb** [comment]
hey @diogenesbrussels 

I think this is an issue with how quotes are escaped when Budibase is running SQL queries. As a workaround I was able to get this working on my Postgres DB by using 'Is in' and then leaving the value blank. Hope this helps!

<img width="988" alt="Screenshot 2023-01-17 at 15 30 07" src="https://user-images.githubusercontent.com/110921612/212940812-b7b572a8-f923-4657-be72-d0f82335d7a3.png">

**Reply by diogenesbrussels** [reply]
Hey @andz-bb,

Thank you for the quick reply, using the IS IN with an empty value throws a SQL syntax error.
`You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near
`

**Reply by andz-bb** [reply]
hi @diogenesbrussels 

I spun up a MariaDB instance to test and I was getting the same result mentioned above in my testing with postgres.
<img width="2266" alt="Screenshot 2023-01-18 at 14 14 23" src="https://user-images.githubusercontent.com/110921612/213194427-ac981b1c-1b6d-4f76-9b0d-cf1f46f5ab1a.png">

What version of MariaDB are you using and what is the datatype of the field you are trying to filter on? Thanks
